vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-integration-tests/src/main/java/com/vaadin/flow/component/spreadsheet/tests/fixtures/SimpleCustomEditorFixture.java

```diff
@@ -1,57 +1,190 @@
+                    componentType.getColumnIndex(),
+                    componentType.getHeaderText());
+        }
+        spreadsheet.setSpreadsheetComponentFactory(new CustomEditorFactory());
+    }
+}
+
+class CustomEditorFactory implements SpreadsheetComponentFactory {
+
+    private TextField textField;
+
+    private TextArea textArea;
+
+    private Checkbox checkbox;
+
+    private DatePicker datePicker;
+
+    private ComboBox<String> comboBox;
+
+    @Override
+    public Component getCustomComponentForCell(Cell cell, int rowIndex,
+            int columnIndex, Spreadsheet spreadsheet, Sheet sheet) {
+        return null;
+    }
+
+    @Override
+    public Component getCustomEditorForCell(Cell cell, final int rowIndex,
+            final int columnIndex, final Spreadsheet spreadsheet, Sheet sheet) {
+        EditorType editorType = EditorType.getEditorTypeByIndex(rowIndex,
+                columnIndex);
+        if (editorType == null) {
+            return null;
+        }
+        Component editor = getCustomEditor(editorType, rowIndex, columnIndex,
+                spreadsheet);
+        if (cell != null) {
+            ((HasValue) editor)
+                    .setValue(getCellValueForEditor(editorType, cell));
+        }
+        return editor;
+    }
 
-        // spreadsheet.setSpreadsheetComponentFactory(new ComponentFactory());
-
-        List<Cell> cellsToRefresh = new ArrayList<Cell>();
-        cellsToRefresh.add(spreadsheet.createCell(0, 0, 1));
-
-        // this has no effect. sheet must be scrolled to "redisplay" the cell.
-        spreadsheet.refreshCells(cellsToRefresh);
-    }
-
-    // TODO Re-enable when SpreadsheetComponentFactory is done
-    // @SuppressWarnings("serial")
-    // public static class ComponentFactory
-    // implements SpreadsheetComponentFactory {
-    //
-    // private final TextField textField = new TextField("");
-    //
-    // @Override
-    // public Component getCustomComponentForCell(Cell cell, int rowIndex,
-    // int columnIndex, Spreadsheet spreadsheet, Sheet sheet) {
-    // return null;
-    // }
-    //
-    // @Override
-    // public Component getCustomEditorForCell(Cell cell, int rowIndex,
-    // int columnIndex, Spreadsheet spreadsheet, Sheet sheet) {
-    // if (cell != null) {
-    // // textField.focus();
-    // return textField;
-    // }
-    // return null;
-    // }
-    //
-    // @Override
-    // public void onCustomEditorDisplayed(Cell cell, int rowIndex,
-    // int columnIndex, Spreadsheet spreadsheet, Sheet sheet,
-    // Component customEditor) {
-    // if (cell != null) {
-    // String cellValue = spreadsheet.getCellValue(cell);
-    // textField.setValue(cellValue);
-    // }
-    // }
-    // }
+    @Override
+    public void onCustomEditorDisplayed(Cell cell, int rowIndex,
+            int columnIndex, Spreadsheet spreadsheet, Sheet sheet,
+            Component customEditor) {
+        // TODO revisar - what should be done here?
```

Comment:
I guess the idea is that the custom editor is created in `getCustomEditorForCell` and updated in `onCustomEditorDisplayed` (whenever the field becomes visible), the way it was in [V8 version of this fixture](https://github.com/vaadin/spreadsheet/blob/master/vaadin-spreadsheet/src/test/java/com/vaadin/addon/spreadsheet/test/fixtures/SimpleCustomEditorFixture.java#L39-L57)

According to the [documentation](https://github.com/vaadin/spreadsheet/blob/master/vaadin-spreadsheet/src/main/java/com/vaadin/addon/spreadsheet/SpreadsheetComponentFactory.java#L133-L155), "The purpose of this method is to make it possible to share the same editor component instance between multiple cells"

Reply:
Removed TODO comment. Do you think that there is any value in adding tests about this part?

Reply:
As it is, `onCustomEditorDisplayed` seems like a completely untested feature. I think it would be enough if we moved the `setValue` line from `getCustomEditorForCell` to `onCustomEditorDisplayed`

Reply:
Addressed in the latest commit.